src/canisters/frontend/ic-asset/src/sync.rs

```diff
@@ -856,4 +864,82 @@ mod test_gathering_asset_descriptors_with_tempdir {
         asset_descriptors.sort_by_key(|v| v.source.clone());
         assert_eq!(dbg!(asset_descriptors), expected_asset_descriptors);
     }
+
+    #[test]
+    /// Test if the content of `.well-known` is included by default.
+    fn include_well_known_directory_by_default() {
+        let files = HashMap::from([
+            // additional, non-dot dirs and files
+            (Path::new(".well-known/file").to_path_buf(), "".to_string()),
+        ]);
+
+        let assets_temp_dir = create_temporary_assets_directory(files).unwrap();
+        let assets_dir = assets_temp_dir.path().canonicalize().unwrap();
+        let mut asset_descriptors = dbg!(gather_asset_descriptors(&[&assets_dir]).unwrap());
+
+        let mut expected_asset_descriptors = vec![
+            AssetDescriptor::default_from_path(&assets_dir, "file"),
+            AssetDescriptor::default_from_path(&assets_dir, ".well-known/file"),
+        ];
+
+        expected_asset_descriptors.sort_by_key(|v| v.source.clone());
+        asset_descriptors.sort_by_key(|v| v.source.clone());
+        assert_eq!(asset_descriptors, expected_asset_descriptors);
```

Comment:
Since this is tracking inclusion/exclusion of specific files, do you think it would be more clear here, rather than creating asset descriptors and sorting them, to instead check that asset_descriptors contains (or does not contain) assets by key?

In other words, something like this
```
        let asset_keys = asset_descriptors.iter().map(|v| v.key.clone()).collect::<Vec<_>>();
        assert!(asset_keys.contains(".well-known/file".into()));
```